src/main/java/pl/javastart/task/Main.java

```diff
@@ -11,6 +22,56 @@ public static void main(String[] args) {
 
     public void run(Scanner scanner) {
         // uzupełnij rozwiązanie. Korzystaj z przekazanego w parametrze scannera
+        System.out.println("Podaj datę:");
+        String userInput = scanner.nextLine();
+
+        TimeZone localTimeZone = TimeZone.getDefault();
+
+        List<String> listOfPatterns = Arrays.asList("yyyy-MM-dd HH:mm:ss", "dd.MM.yyyy HH:mm:ss", "yyyy-MM-dd");
+
+        boolean patternCheck = false;
+        for (String pat : listOfPatterns) {
+            try {
+                DateTimeFormatter pattern = DateTimeFormatter.ofPattern(pat);
+                TemporalAccessor temporalAccessor = pattern.parse(userInput);
+                LocalDateTime localDateTime;
+                DateTimeFormatter formatter;
+                if (pat.equals("yyyy-MM-dd")) {
+                    formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                    LocalDate localDate = LocalDate.from(temporalAccessor);
+                    System.out.print("Czas lokalny: ");
+                    System.out.println(localDate.format(formatter) + " 00:00:00");
+                    localDateTime = localDate.atTime(0, 0, 0);
+                    formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+                } else {
```

Comment:
Skoro sama data, bez czasu, jest specjalnym przypadkiem, to nie powinno się to odbywać w pętli, w której z założenia powtarzamy ten sam kod dla wielu elementów